crates/ethereum-rpc/src/trace.rs

```diff
@@ -237,10 +237,17 @@ fn get_traces_with_requested_tracer_and_config(
                         }
                         _ => {
                             traces.into_iter().for_each(|trace| {
-                                if let GethTrace::CallTracer(call_frame) = trace {
+                                if let TraceResult::Success {
+                                    result: GethTrace::CallTracer(call_frame),
+                                    tx_hash,
+                                } = trace
+                                {
                                     let new_call_frame =
                                         apply_call_config(call_frame.clone(), call_config);
-                                    new_traces.push(GethTrace::CallTracer(new_call_frame));
+                                    new_traces.push(TraceResult::new_success(
```

Comment:
We never push errored traces?

Reply:
In the case of a tracing error, I think it is logical to just the return Err instead of pushing TraceResult::Error, since we need to abort when a tracing error occurs, tx are dependent on each other. This is in line with the discussion here: https://github.com/paradigmxyz/reth/pull/7837#discussion_r1577739078